sources/PreferenceInfo.m

```diff
@@ -42,10 +44,10 @@ - (void)setObserver:(void (^)())observer {
     _observer = [observer copy];
     // Call the observer after a delayed perform so that the current profile can be set and then the
     // control's value gets initialized.
-    [self performSelector:@selector(callObserver) withObject:nil afterDelay:0];
+    [self _callObserver];
```

Comment:
I'm not convinced we can get rid of the delayed perform. Every observer would need to be analyzed to ensure no assumptions were broken. I'm not crazy about using a delayed perform, either, but it is nice and simple. One alternative would be to run all the observers after the prefs window finishes loading, perhaps by notification.